### ISSUE TITLE
[feature request] Add custom checks

### ISSUE THREAD

**Issue by hoshsadiq** [adrienverge/yamllint#123]
Hey all,

It would be nice to have an extra option to allow a custom rules import path that can be added to the rules. For example:

```bash
$ yamllint --extra-rules ./yaml-rules myfile.yaml
```

This would allow a user to add python scripts to `./yaml-rules` with a check method with the correct signature. These rules could then be used the same way the default rules are used (i.e. disable in `.yamllint`, or in comments etc).

I could try and have a go at implementing such a feature, but I figured I'd ask the maintainers first so I don't waste my time in case you don't want such a feature (or perhaps you have opinions on how it should be implemented).

This feature is similar to how https://github.com/awslabs/cfn-python-lint has an option to add additional rules (see https://github.com/awslabs/cfn-python-lint/blob/master/docs/getting_started/rules.md).

**Comment by adrienverge** [comment]
Hi Hosh,

Can I ask what kind of custom rule you'd like to have? Maybe it could be integrated upstream instead of having an `--extra-rules` feature?

Anyway the pluggable rules use case makes sense, but in order not to waste time I suggest you describe how you'd like to implement it! Also you should look at https://github.com/adrienverge/yamllint/pull/19 by @pdericson.

Cheers

**Comment by hoshsadiq** [comment]
My use case is custom domain checks. E.g. we have a governance within our yaml that requires certain values for certain elements amongst other things. We want these rules to be enabled/disabled in line as well as globally per file. #19's usecase makes more sense to be part of the project, however, our usecase is quite specific to our custom yaml models. I will write a quick proposal on how I'd like to implement this. I will take into account #19 as he's done a huge chunk of the work, so I'll take into account your comments on that PR.

**Comment by ChristianWeissCG** [comment]
Just want to show my mental support to this request. 
I would love to see a `--load-additional-rules`.

Maybe we can change the titel to `Use custom checks / rules`.

**Comment by toffiecoffie** [comment]
Exactly what I am looking to do....

**Comment by adrienverge** [comment]
@toffiecoffie, maybe you could post here instead of creating two new issues? #172 #173 

About your problem, I suggest your take inspiration from a commit that adds a new rule (see for instance commit 1543d0e "New rule key-ordering") and make sure you do all what's needed.

**Comment by ChristianWeissCG** [comment]
I do not think that it is required to add a new rule to the default ruleset (as disabled):
https://github.com/adrienverge/yamllint/commit/1543d0e43556cd5165a7e785395964f9badfa404#diff-80ac6fe01633dda82861672967172ec9R42
 because it would not get respected on exection of default ruleset when not defined in that ruleset at all.

**Comment by dozhang** [comment]
I also wanted to support this old feature request, which support of customized rules. My use case is to adding domain specific static checks. Something like: if element attribute A is set to True, then B should also set to True, etc. 
Yes this could be implemented in the application itself, but it would be much easier to plug it in the lint, which can also support file comments in Github, etc. 
I also checked the related tickets, seems the maintainers do not like this feature, but I haven't see a good reason.
